[
    {
        "content": "<p>I am trying out python-dvuploader for uploading large datasets. Is there a public instance available for testing that support direct uploads? The ones I have tested fall back to the native api. <span class=\"user-mention\" data-user-id=\"599841\">@Jan Range</span> , do you have suggestions?</p>",
        "id": 503060261,
        "sender_full_name": "Mattias de Hollander",
        "timestamp": 1741016989
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"883325\">@Mattias de Hollander</span> thanks for reaching out! Unfortunately, there is no public direct upload Dataverse instance to my knowledge. If it's solely for testing purposes, I recommend using Docker to host it locally. Would this be an option for you? Happy to help setting it up <span aria-label=\"smile\" class=\"emoji emoji-1f642\" role=\"img\" title=\"smile\">:smile:</span></p>",
        "id": 503061287,
        "sender_full_name": "Jan Range",
        "timestamp": 1741017250
    },
    {
        "content": "<p>Ok, than I don't look any further. And thanks for the fast reply and developing this client/library. Running it locally with docker is possible for sure. I give that a try.</p>",
        "id": 503064585,
        "sender_full_name": "Mattias de Hollander",
        "timestamp": 1741017998
    },
    {
        "content": "<p>Thanks a lot! Let me know if you need any help <span aria-label=\"smile\" class=\"emoji emoji-1f642\" role=\"img\" title=\"smile\">:smile:</span></p>",
        "id": 503064882,
        "sender_full_name": "Jan Range",
        "timestamp": 1741018055
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"883325\">@Mattias de Hollander</span> the default Dataverse dev environment includes LocalStack with direct upload enabled: <a href=\"https://guides.dataverse.org/en/6.5/developers/dev-environment.html#quickstart\">https://guides.dataverse.org/en/6.5/developers/dev-environment.html#quickstart</a></p>\n<p>This line: <a href=\"https://github.com/IQSS/dataverse/blob/v6.5/docker-compose-dev.yml#L40\">https://github.com/IQSS/dataverse/blob/v6.5/docker-compose-dev.yml#L40</a></p>",
        "id": 503080346,
        "sender_full_name": "Philip Durbin ðŸš€",
        "timestamp": 1741021752
    },
    {
        "content": "<p>Also a note, when using LocalStack, set the <code>DVUPLOADER_TESTING</code> environment variable to <code>True</code>. Otherwise, the received URLs internally won't work, since these are aliased by Docker. The URL <code>http://localstack/...</code> will then be converted to <code>http://localhost/...</code>.</p>",
        "id": 503081398,
        "sender_full_name": "Jan Range",
        "timestamp": 1741022046
    },
    {
        "content": "<p><a href=\"https://github.com/gdcc/python-dvuploader?tab=readme-ov-file#development\">https://github.com/gdcc/python-dvuploader?tab=readme-ov-file#development</a></p>",
        "id": 503081509,
        "sender_full_name": "Jan Range",
        "timestamp": 1741022083
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"883325\">@Mattias de Hollander</span> so, I just learned that direct upload is enabled on <a href=\"https://beta.dataverse.org\">https://beta.dataverse.org</a> (on the default store) so you could try your test there.</p>",
        "id": 503083223,
        "sender_full_name": "Philip Durbin ðŸš€",
        "timestamp": 1741022635
    },
    {
        "content": "<p>Good to hear direct upload is enabled on the beta dataverse website. Has been a while since I worked on dataverse and was not aware of this instance. However, I can register, but I think I need an elevated role to be able to add data, or create a new dataverse first. Can someone do that? Thanks.</p>\n<p>In the mean time I give spinning up a docker instance a try.</p>",
        "id": 503221676,
        "sender_full_name": "Mattias de Hollander",
        "timestamp": 1741081494
    },
    {
        "content": "<p>Are you able to create a dataverse (we're starting to call these \"collections\" by the way).</p>",
        "id": 503266036,
        "sender_full_name": "Philip Durbin ðŸš€",
        "timestamp": 1741094572
    },
    {
        "content": "<p>Not at <a href=\"http://beta.dataverse.org\">beta.dataverse.org</a>. Good naming update :)</p>",
        "id": 503270059,
        "sender_full_name": "Mattias de Hollander",
        "timestamp": 1741095618
    },
    {
        "content": "<p><span class=\"user-mention\" data-user-id=\"883325\">@Mattias de Hollander</span> it says, \"Anyone with a Dataverse account can add sub dataverses\":</p>\n<p><a href=\"/user_uploads/53090/_flbAs_K5PBDB7yMLR9dDuKK/Screenshot-2025-03-04-at-9.10.55AM.png\">Screenshot 2025-03-04 at 9.10.55â€¯AM.png</a></p>\n<div class=\"message_inline_image\"><a href=\"/user_uploads/53090/_flbAs_K5PBDB7yMLR9dDuKK/Screenshot-2025-03-04-at-9.10.55AM.png\" title=\"Screenshot 2025-03-04 at 9.10.55â€¯AM.png\"><img data-original-content-type=\"image/png\" data-original-dimensions=\"3162x1948\" src=\"/user_uploads/thumbnail/53090/_flbAs_K5PBDB7yMLR9dDuKK/Screenshot-2025-03-04-at-9.10.55AM.png/840x560.webp\"></a></div>",
        "id": 503277534,
        "sender_full_name": "Philip Durbin ðŸš€",
        "timestamp": 1741097491
    },
    {
        "content": "<p>You don't see an \"Add Data\" button?</p>",
        "id": 503277762,
        "sender_full_name": "Philip Durbin ðŸš€",
        "timestamp": 1741097535
    },
    {
        "content": "<p>I do see that button. Here is my view:<br>\n<a href=\"/user_uploads/53090/QGhNIZRmLw_minzI-XVeMoXv/image.png\">image.png</a></p>\n<div class=\"message_inline_image\"><a href=\"/user_uploads/53090/QGhNIZRmLw_minzI-XVeMoXv/image.png\" title=\"image.png\"><img data-original-content-type=\"image/png\" data-original-dimensions=\"1313x1011\" src=\"/user_uploads/thumbnail/53090/QGhNIZRmLw_minzI-XVeMoXv/image.png/840x560.webp\"></a></div><p>And the response for both options is this:<br>\n<a href=\"/user_uploads/53090/F0OzP7x0TlJ2P0sL85WKaE0K/image.png\">image.png</a></p>\n<div class=\"message_inline_image\"><a href=\"/user_uploads/53090/F0OzP7x0TlJ2P0sL85WKaE0K/image.png\" title=\"image.png\"><img data-original-content-type=\"image/png\" data-original-dimensions=\"1323x413\" src=\"/user_uploads/thumbnail/53090/F0OzP7x0TlJ2P0sL85WKaE0K/image.png/840x560.webp\"></a></div>",
        "id": 503281371,
        "sender_full_name": "Mattias de Hollander",
        "timestamp": 1741098416
    },
    {
        "content": "<p>500 error?!? D'oh! <img alt=\":doh:\" class=\"emoji\" src=\"https://avatars.zulip.com/53090/emoji/images/343009b9.png\" title=\"doh\"></p>",
        "id": 503281490,
        "sender_full_name": "Philip Durbin ðŸš€",
        "timestamp": 1741098445
    },
    {
        "content": "<p><code>org.postgresql.util.PSQLException: ERROR: column \"displayoncreate\" does not exist</code></p>",
        "id": 503281844,
        "sender_full_name": "Philip Durbin ðŸš€",
        "timestamp": 1741098528
    },
    {
        "content": "<p>Shoot. We're working a fix for this. Please disregard what I said. Please don't use the beta server until this is fixed. I'll let you know!</p>",
        "id": 503281974,
        "sender_full_name": "Philip Durbin ðŸš€",
        "timestamp": 1741098561
    },
    {
        "content": "<p>No problem. Happy when it is there, otherwise local options take more time but also work out. Thanks anyways.</p>",
        "id": 503282370,
        "sender_full_name": "Mattias de Hollander",
        "timestamp": 1741098647
    },
    {
        "content": "<p>If you create a collection on <a href=\"https://demo.dataverse.org\">https://demo.dataverse.org</a> I can point it to store with S3 direct upload configured. Lemme know.</p>",
        "id": 503284025,
        "sender_full_name": "Philip Durbin ðŸš€",
        "timestamp": 1741099076
    },
    {
        "content": "<p>That would be great. Here is one: <a href=\"https://demo.dataverse.org/dataverse/mhtest1\">https://demo.dataverse.org/dataverse/mhtest1</a></p>",
        "id": 503299093,
        "sender_full_name": "Mattias de Hollander",
        "timestamp": 1741102654
    },
    {
        "content": "<p>Ok, please try it now.</p>",
        "id": 503305182,
        "sender_full_name": "Philip Durbin ðŸš€",
        "timestamp": 1741104140
    },
    {
        "content": "<p>Thanks! I give it a try tomorrow and let you know.</p>",
        "id": 503310816,
        "sender_full_name": "Mattias de Hollander",
        "timestamp": 1741105568
    },
    {
        "content": "<p>Super, it works now with direct upload support. I will start exploring it. Thanks a lot for the support.</p>",
        "id": 503449710,
        "sender_full_name": "Mattias de Hollander",
        "timestamp": 1741163175
    },
    {
        "content": "<p>Yep, sure.</p>",
        "id": 503495350,
        "sender_full_name": "Philip Durbin ðŸš€",
        "timestamp": 1741176581
    },
    {
        "content": "<p>I've started testing the S3 direct upload support, thanks for making this possible. If you have any ideas regarding my follow-up question about setting parameters for direct upload with python-dvuploader (<a href=\"#narrow/channel/432390-large-data/topic/setting.20parameters.20for.20direct.20upload.20with.20python-dvuploader/with/504607881\">https://dataverse.zulipchat.com/#narrow/channel/432390-large-data/topic/setting.20parameters.20for.20direct.20upload.20with.20python-dvuploader/with/504607881</a>), I'd appreciate your input.</p>",
        "id": 504858615,
        "sender_full_name": "Mattias de Hollander",
        "timestamp": 1741703490
    }
]